main.tf

```diff
@@ -1,13 +1,178 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+# ----------------------------------------------------------------------------------------------------------------------
+# Enable and configure AWS Config
+# ----------------------------------------------------------------------------------------------------------------------
+module "aws_config_label" {
+  source  = "cloudposse/label/null"
+  version = "0.21.0"
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  attributes = ["config"]
+  context    = module.this.context
+}
+resource "aws_config_configuration_recorder" "recorder" {
+  count    = module.this.enabled ? 1 : 0
+  name     = module.aws_config_label.id
+  role_arn = local.create_iam_role ? module.iam_role[0].arn : var.iam_role_arn
+  recording_group {
+    all_supported                 = true
+    include_global_resource_types = var.include_global_resource_types
+  }
+}
+
+resource "aws_config_delivery_channel" "channel" {
+  count          = module.this.enabled ? 1 : 0
+  name           = module.aws_config_label.id
+  s3_bucket_name = module.aws_config_storage[0].bucket_id
+  s3_key_prefix  = "foo"
```

Comment:
what is `foo` here?
If we need d=some value, let's parameterize it.

Reply:
Foo was a leftover from testing. I should probably add an optional variable that allows the user to specify the prefix.

Reply:
24367ecf294fb1392c6fc2ea37abe03dded49f68